### DISCUSSION TITLE
[Feature request] Mobile APP with 2 Server URL (internal and external)

### DISCUSSION THREAD

**Discussion by mjimeneznet** [immich-app/immich#9550] [General]
I'm a newcomer to Immich and during the configuration of the iOS app, I noticed it only requests a single server endpoint for connection.

This differs from my experience with Home Assistant, where the companion app allows for the configuration of two distinct endpoints - one for local use and another for external access.

The advantage of this dual-endpoint setup is its adaptability. When you're at home and connected to your Wi-Fi, the app recognizes your network and utilizes the local URL for connection, ensuring a straightforward and efficient process.

However, when you're away from home, either connected to an unfamiliar Wi-Fi or using a 4G/5G network, the app switches to the external endpoint. This endpoint can be configured according to your preference, be it a Cloudflare tunnel, port forwarding, etc.

This feature is particularly beneficial for those who self-host applications. It allows for the use of a local URL when at home and a Cloudflare tunnel or a port-forwarded one when outside, using a DDNS, etc (many options) This flexibility optimizes connection efficiency and security, adapting to your location and network conditions.

**Comment by bo0tzz** [comment]
#308
